src/V3Ast.cpp

```diff
@@ -1082,26 +1082,97 @@ void AstNode::dumpPtrs(std::ostream& os) const {
 
 void AstNode::dumpTree(std::ostream& os, const string& indent, int maxDepth) const {
     static int s_debugFileline = v3Global.opt.debugSrcLevel("fileline");  // --debugi-fileline 9
-    os << indent << " " << this << '\n';
-    if (debug() > 8) {
-        os << indent << "     ";
-        dumpPtrs(os);
-    }
-    if (s_debugFileline >= 9) os << fileline()->warnContextSecondary();
-    if (maxDepth == 1) {
-        if (op1p() || op2p() || op3p() || op4p()) os << indent << "1: ...(maxDepth)\n";
-    } else {
+    if (v3Global.opt.dumpTreeDot()) {
+        os << this;
+        if (op1p()) {
+            if (op1p()->typeName() != "BASICDTYPE") {
+                os << "\tn" << cvtToHex(this) << " -> n" << cvtToHex(op1p()) << " ["
+                   << "fontsize=8 label=\"" << "op1" << "\""
+                   << " weight=" << 1 << " color=" << "red" << "];\n";
+            }
+        }
         for (const AstNode* nodep = op1p(); nodep; nodep = nodep->nextp()) {
+            if (nodep->typeName() == "BASICDTYPE") { break; }
             nodep->dumpTree(os, indent + "1:", maxDepth - 1);
+            if (nodep->nextp()) {
+                os << "\tn" << cvtToHex(nodep) << " -> n" << cvtToHex(nodep->nextp()) << " ["
+                   << "fontsize=8 label=\"" << "next" << "\""
+                   << " weight=" << 1 << " color=" << "red" << "];\n";
+                os << "\t{rank=same; n" << cvtToHex(nodep) << ", n" << cvtToHex(nodep->nextp())
+                   << "}\n";
+            }
+        }
```

Comment:
This is repeated 4 times with only the op number being different. Lets's factor this into a helper function.